src/corehost/cli/fx_reference.cpp

```diff
@@ -23,39 +23,60 @@ bool fx_reference_t::is_roll_forward_compatible(const fx_ver_t& other) const
     }
 
     // Verify major roll forward
+    static_assert(
+        roll_forward_option::LatestMajor > roll_forward_option::Major,
+        "Code assumes ordering of roll-forward options from least restrictive to most restrictive");
```

Comment:
I would put all the static_asserts of this category together in one code block so that it was clear that the full order was tested exhaustively.